pkg/controller/shared_state.go

```diff
@@ -1175,9 +1176,14 @@ func (c *SharedState) createReplicas(ctx context.Context, remainingReplicas int,
 			//Push to queue the failed replica number
 			request := ReplicaRequest{VolumeName: volumeName, Priority: remainingReplicas}
 			c.priorityReplicaRequestsQueue.Push(ctx, &request)
-			return err
+			// don't return err yet, we can still try the next node
+			continue
```

Comment:
This means that as long as there are nodes we will attempt to create replicas. If all fail (or one fails when the last node is attempted), we will not requeue an attempt to retry (lines 1189..1195 below). I do not believe we should continue on error to create the `AzVolumeAttachment` here.

Reply:
Oh I see, we would requeue an attempt on failure (line 1175..1179 above).

Still, it seems like we could simplify a bit and only decrement `remainingReplicas` on success (exiting early when we reach zero as you've added below). That way, we can rely on `remainingReplicas > 0` to requeue the retry as long as there are more replicas needed. This would eliminate the need to separately requeue on failure to create the `AzVolumeAttachment`.

Reply:
Because the function getNodesForReplica() will return a full list of nodes, so if the attachment fails here, we can still try the next node until remainingReplicas is 0.